### ISSUE TITLE
Package it as a Snap

### ISSUE THREAD

**Issue by heidiwenger** [hongquan/CoBang#43]
For an easy installation clicking in Software in Ubuntu, which is the first Linux distro for many new users now coming to Linux.

Thanks a million if this could be possible!

**Comment by hongquan** [comment]
I will look into it. It may take sometimes for me to learn how to package as snap. Unfortunately my health is not good these days so I cannot spend time for it yet.

**Comment by heidiwenger** [comment]
> I will look into it. It may take sometimes for me to learn how to package as snap. Unfortunately my health is not good these days so I cannot spend time for it yet.

Thank you so much for being so kind, but health absolutely first, yes. 

Here's a "Zero to snap in (much less than actually) 60 minutes" tutorial video by "Mr. Snap himself" Alan Pope (former Canonical) if it's of any help: https://www.youtube.com/watch?v=ikoUByA5BLU

Also there's one from his blog: https://popey.com/blog/2021/02/lets-go-snapping/

**Comment by heidiwenger** [comment]
Oh i just remembered, there is also this spot for snaps (requests). People there can also help. I put CoBang there as a new request: https://forum.snapcraft.io/c/snap-requests/21

**Comment by vrement** [comment]
> I will look into it. It may take sometimes for me to learn how to package as snap. Unfortunately my health is not good these days so I cannot spend time for it yet.

Hopefully feeling better. I see there is the milestone of CoBang 1.0 released, hoping there'd be a snap package now for simple one click install on Ubuntu which is so popular always.

**Comment by hongquan** [comment]
Thanks, will find some times to do.

**Comment by vrement** [comment]
> [@vrement](https://github.com/vrement) Unfortunately, Snap `core24` base doesn't contain enough dependencies to build:
> 
>     * It doesn't have [`gstreamer1.0-gtk4`](https://packages.ubuntu.com/oracular/gstreamer1.0-gtk4).
> 
>     * Tried building [`gst-plugins-rs`](https://github.com/GStreamer/gst-plugins-rs) from source but `core24` contains older Cargo than requirements.
> 
> 
> Maybe we need to wait for Canonical to release `core26`.

Yes, maybe that would be for the best. Snapcrafters can also be communicated (with possible requests on this) here: https://forum.snapcraft.io/c/snapcraft/13

**Comment by hongquan** [comment]
Asked in [Snapcraft forum](https://forum.snapcraft.io/t/snap-with-dependencies-existent-on-24-10/45494), but Oliver Grawert suggested to backport the missing packages to Ubuntu 24.04. It is a very difficult task when it is related to Rust.

**Comment by vrement** [comment]
> Asked in [Snapcraft forum](https://forum.snapcraft.io/t/snap-with-dependencies-existent-on-24-10/45494), but Oliver Grawert suggested to backport the missing packages to Ubuntu 24.04. It is a very difficult task when it is related to Rust.

I sure hope it is doable, though 👍
